src/javascript/class-scanner.ts

```diff
@@ -335,6 +347,221 @@ interface CustomElementDefinition {
+
+    const annotation = getJSDocAnnotationForNode(node);
+
+    if (babel.isFunctionExpression(expr.right)) {
+      const method = this._createMethodFromExpression(leftProperty.name,
+        expr.right, annotation);
+      if (method) {
+        this._addMethodToClass(cls, method);
+      }
+    } else {
+      const prop = this._createPropertyFromExpression(leftProperty.name,
+        expr, annotation);
+      if (prop) {
+        this._addPropertyToClass(cls, prop);
+      }
+    }
+  }
+
+  private _addMethodToClass(cls: string, member: ScannedMethod) {
+    let classMembers = this.members.get(cls);
+    if (!classMembers) {
+      classMembers = {
+        methods: new Map<string, ScannedMethod>(),
+        properties: new Map<string, ScannedProperty>()
+      };
+      this.members.set(cls, classMembers);
+    }
+    classMembers.methods.set(member.name, member);
+  }
+
+  private _addPropertyToClass(cls: string, member: ScannedProperty) {
+    let classMembers = this.members.get(cls);
+    if (!classMembers) {
+      classMembers = {
+        methods: new Map<string, ScannedMethod>(),
+        properties: new Map<string, ScannedProperty>()
+      };
+      this.members.set(cls, classMembers);
+    }
+    classMembers.properties.set(member.name, member);
+  }
+
+  private _createMemberFromMemberExpression(node: babel.ExpressionStatement) {
+    if (!babel.isMemberExpression(node.expression)) {
+      return;
+    }
+
+    const left = node.expression.object;
+
+    // we only want `something.prototype.member`
+    if (!babel.isIdentifier(node.expression.property) ||
+        !babel.isMemberExpression(left) ||
+        getIdentifierName(left.property) !== 'prototype') {
+      return;
+    }
+
+    const cls = getIdentifierName(left.object);
+
+    if (!cls) {
+      return;
+    }
+
+    const annotation = getJSDocAnnotationForNode(node);
+
+    if (jsdoc.hasTag(annotation, 'function')) {
+      const method = this._createMethodFromExpression(node.expression.property.name,
+        node.expression, annotation);
+      if (method) {
+        this._addMethodToClass(cls, method);
+      }
+    } else {
+      const prop = this._createPropertyFromExpression(node.expression.property.name,
+        node.expression, annotation);
+      if (prop) {
+        this._addPropertyToClass(cls, prop);
+      }
+    }
+  }
+
+  private _createPropertyFromExpression(
+      name: string, node: babel.AssignmentExpression|babel.MemberExpression, jsdocAnn?: jsdoc.Annotation) {
+    let description;
+    let type;
+    let privacy: Privacy = 'public';
+    let readOnly = false;
+    const sourceRange = this._document.sourceRangeForNode(node)!;
+    const warnings: Warning[] = [];
+
+    if (jsdocAnn) {
+      const typeTag = jsdoc.getTag(jsdocAnn, 'type');
+      if (typeTag && typeTag.type) {
+        type = doctrine.type.stringify(typeTag.type);
+      }
+      description = getDescription(jsdocAnn);
+      privacy = getOrInferPrivacy(name, jsdocAnn);
```

Comment:
We should use get or infer privacy even if we don't have a jsdoc, so that we pay attention to the name of the property